### PR TITLE
Registrator guidelines for auto-merging

### PR DESCRIPTION
**Pull request by paraynaud**
add compats in Project.toml

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -16,6 +16,10 @@ StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
+julia = "^1.6.0"
 LDLFactorizations = "0.8"
 PrettyTables = "1"
 StatsBase = "0.33"
```
Comment (dpo) [inline Project.toml:22]: What are the two above needed for?

Comment (paraynaud) [reply Project.toml:22]: I use `StatsBase` for the `sample()` function, which is a `rand()` without repetetion.
It is used to define random partitioned structures.

`PrettyTables` is currently not used.

Comment (dpo) [reply Project.toml:21]: Ok. What's the use of defining random structures? If it's for the unit tests, then StatsBase should only be an "extra" dependency (to be used in the unit tests).

Comment (paraynaud) [reply Project.toml:21]: It is used in the definition of the partitioned structures itself.
These random structures are mainly defined for tests.

Comment (dpo) [reply Project.toml:21]: Ok, I think we also need to open an issue to move StatsBase to the "Extra" dependencies in the future.

**Comment by paraynaud** [comment]
I think we are good to merge this PR and trigger the registrator bot.

File: Project.toml [hunk 2]
```diff
@@ -16,6 +16,9 @@ StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
+julia = "^1.6.0"
 LDLFactorizations = "0.8"
-PrettyTables = "1"
 StatsBase = "0.33"
+BenchmarkTools = "1"
+LinearOperators = "2"
+Revise = "3"
```
Comment (dpo) [inline Project.toml:24]: Does this package use Revise?!?!

File: Project.toml [hunk 3]
```diff
@@ -16,6 +16,9 @@ StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
+julia = "^1.6.0"
 LDLFactorizations = "0.8"
-PrettyTables = "1"
 StatsBase = "0.33"
+BenchmarkTools = "1"
```
Comment (dpo) [inline Project.toml:22]: I'm also surprised you would be using BenchmarkTools internally!!! 🤔

**Review by dpo — CHANGES_REQUESTED** [review]
Also your unit tests are importing packages that are not listed as "Extra" in `Project.toml`: StatsBase, LinearOperators, SparseArrays, LinearAlgebra, ... and Test!

**Comment by dpo** [comment]
I think this is where you use StatsBase: https://github.com/JuliaSmoothOptimizers/PartitionedStructures.jl/blob/3ed9f43d209f62c81efefa35fd87872db462725f/src/P_vec/elemental_ev.jl#L43

Why does `new_eev` need to return random vectors? Shouldn't it return un-initialized vectors? Something like
```julia
Elemental_elt_vec(Vector{T}(undef, nᵢ))
```

**Comment by paraynaud** [comment]
> Also your unit tests are importing packages that are not listed as "Extra" in `Project.toml`: StatsBase, LinearOperators, SparseArrays, LinearAlgebra, ... and Test!

Except Test, those packages are used in the source code.
Do I need to duplicate them in the [extras] field?

**Comment by paraynaud** [comment]
> I think this is where you use StatsBase:
> 
> https://github.com/JuliaSmoothOptimizers/PartitionedStructures.jl/blob/3ed9f43d209f62c81efefa35fd87872db462725f/src/P_vec/elemental_ev.jl#L43
> 
> Why does `new_eev` need to return random vectors? Shouldn't it return un-initialized vectors? Something like
> 
> ```julia
> Elemental_elt_vec(Vector{T}(undef, nᵢ))
> ```

I made it like this, but an `undef` vector in place of a `rand(n)` vector make more sense.
I open an issue.

**Comment by paraynaud** [comment]
How do I run unit tests locally if I don't have `Test ` in my `[deps]?

**Comment by paraynaud** [comment]
Is this all good @dpo?

**Comment by dpo** [comment]
You have to add the "extra" dependencies as targets. Check the other examples of Project.toml.

**Comment by paraynaud** [comment]
I had rhe `target`.

**Review by dpo — APPROVED** [review]

**Pull request merged**